exercises/michael_turner/ex02_oo_basics/src/shape_sorter.cpp

```diff
@@ -0,0 +1,119 @@
+#include "shape_sorter.h"
+
+#include "shapes.h"
+
+#include <iostream>
+#include <iomanip>
+#include <functional>
+#include <string>
+#include <map>
+#include <vector>
+
+/**
+ * @brief ShapeSorter::ShapeSorter      Reads in the vector of shapes to be sorted
+ * @param v         The vector of shapes
+ */
+ShapeSorter::ShapeSorter(ShapeVector &v): shapes(v){}
+
+
+/**
+ * @brief ShapeSorter::printRank        Prints the rank of shapes (sorted by area or perimeter)
+ *                                                          in decreasing order
+ * @param rank      The shapes in ranking order
+ */
+void ShapeSorter::printRank(std::multimap<double, std::reference_wrapper<Shape> > &rank){
+
+    //Iterate through map in reverse order and print value of area/perimeter, followed by the Shape data
+    for(auto iter = rank.crbegin(); iter != rank.crend(); ++iter){
+        //Print data in columns
+        std::cout <<std::left << std::setw(COL_WIDTH) << iter->first;
+        iter->second.get().printShape();
+    }
+    std::cout << std::endl;
+}
+
+
+/**
+ * @brief ShapeSorter::printSort        Prints the data of the shapes matching a criterium
+ * @param sorted            The selected shapes
+ */
+void ShapeSorter::printSort(ShapeVector &sorted){
+    //Iterate through vector and print each shape's data
+    for(const auto &shape_ref: sorted){
```

Comment:
Doh, I think I suggested the name `shape_ref` (old habit) but actually the convention in Mantid is for camelCase variable names. My bad. No need to change it.